### PR TITLE
X509 Read and profile fix

### PR DESCRIPTION
**Pull request by Adelmar92**
## Description

2 main changes were made in this PR:

- the Project ID now is part of the primary identifiers
- Apikeys, are no longer an input, now they are obtained from a secret named cfn/atlas/profile/{profile}

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by colm-quinn** [comment]
I tyhink you can remove all the `*/DSSTORE` binary files from the PR & add them to .gitignore

File: examples/profile-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    Default: "PrivateKey"
+Resources:
+  AtlasApiKeySecret:
+    Type: 'AWS::SecretsManager::Secret'
+    Properties:
+      Name: !Sub "cfn/atlas/profile/${ProfileName}"
+      Description: MongoDB Atlas API Key
+      #
+      # { "PublicKey" : "xxx", "PrivateKey", "yyy", "OrgId": "zzz" }
+      #
```
Comment (colm-quinn) [inline examples/profile-secret.yaml:24]: [q] Anyway you can include the example in the description test? If multiline/code isn't supported no point, just a nice to have.

File: examples/x509-authentication-db-user/x509-authentication-db-user.json [hunk 2]
```diff
@@ -4,46 +4,53 @@
...
-	  "Default" : "",
+      "Description" : "Certificate (default value is just an example)",
+	  "Default" : "-----BEGIN CERTIFICATE-----\nMIIC4jCCAcoCCQC33wnybT5QZDANBgkqhkiG9w0BAQsFADAyMQswCQYDVQQGEwJV\nSzEPMA0GA1UECgwGQm94eUhRMRIwEAYDVQQDDAlNb2NrIFNBTUwwIBcNMjIwMjI4\nMjE0NjM4WhgPMzAyMTA3MDEyMTQ2MzhaMDIxCzAJBgNVBAYTAlVLMQ8wDQYDVQQK\nDAZCb3h5SFExEjAQBgNVBAMMCU1vY2sgU0FNTDCCASIwDQYJKoZIhvcNAQEBBQAD\nggEPADCCAQoCggEBALGfYettMsct1T6tVUwTudNJH5Pnb9GGnkXi9Zw/e6x45DD0\nRuRONbFlJ2T4RjAE/uG+AjXxXQ8o2SZfb9+GgmCHuTJFNgHoZ1nFVXCmb/Hg8Hpd\n4vOAGXndixaReOiq3EH5XvpMjMkJ3+8+9VYMzMZOjkgQtAqO36eAFFfNKX7dTj3V\npwLkvz6/KFCq8OAwY+AUi4eZm5J57D31GzjHwfjH9WTeX0MyndmnNB1qV75qQR3b\n2/W5sGHRv+9AarggJkF+ptUkXoLtVA51wcfYm6hILptpde5FQC8RWY1YrswBWAEZ\nNfyrR4JeSweElNHg4NVOs4TwGjOPwWGqzTfgTlECAwEAATANBgkqhkiG9w0BAQsF\nAAOCAQEAAYRlYflSXAWoZpFfwNiCQVE5d9zZ0DPzNdWhAybXcTyMf0z5mDf6FWBW\n5Gyoi9u3EMEDnzLcJNkwJAAc39Apa4I2/tml+Jy29dk8bTyX6m93ngmCgdLh5Za4\nkhuU3AM3L63g7VexCuO7kwkjh/+LqdcIXsVGO6XDfu2QOs1Xpe9zIzLpwm/RNYeX\nUjbSj5ce/jekpAw7qyVVL4xOyh8AtUW1ek3wIw1MJvEgEPt0d16oshWJpoS1OT8L\nr/22SvYEo3EmSGdTVGgk3x3s+A0qWAqTcyjr7Q4s/GKYRFfomGwz0TZ4Iw1ZN99M\nm0eo2USlSRTVl7QHRTuiuSThHpLKQQ==\n-----END CERTIFICATE-----",
```
Comment (colm-quinn) [inline examples/x509-authentication-db-user/x509-authentication-db-user.json:8]: Pretty sure people will just keep the default if the example is too real looking. How about?

```suggestion
	  "Default" : "-----BEGIN CERTIFICATE-----\n<CERTIFICATE_HASH>\n-----END CERTIFICATE-----",
```

File: examples/x509-authentication-db-user/x509-authentication-db-user.json [hunk 3]
```diff
@@ -4,46 +4,53 @@
...
     },
     "MonthsUntilExpiration": {
       "Type": "String",
       "Description": "",
       "Default" : 0,
       "ConstraintDescription": ""
+    },
+    "Profile": {
+      "Type": "String",
+      "Description": "MongoDB Atlas Proj Profile Id",
```
Comment (colm-quinn) [inline examples/x509-authentication-db-user/x509-authentication-db-user.json:30]:
```suggestion
      "Description": "MongoDB Atlas Project Profile Id",
```

File: cfn-resources/serverless-instance/cmd/resource/resource.go [hunk 4]
```diff
@@ -104,20 +107,16 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
...
 		return *modelValidation, nil
 	}
 
 	// Create atlas client
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = log.Warnf("Read - error: %+v", err)
-		return progress_events.GetFailedEventByCode(fmt.Sprintf("Error creating mongoDB client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
```
Comment (colm-quinn) [inline cfn-resources/serverless-instance/cmd/resource/resource.go:117]: Missing profile read?

```suggestion
        if currentModel.Profile == nil {
		currentModel.Profile = pointy.String(util.DefaultProfile)
	}
	
	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
```

Comment (colm-quinn) [reply cfn-resources/serverless-instance/cmd/resource/resource.go:117]: Also, wondering should this instead happen in the `util. NewMongoDBClient` as the fallback behaviour is needed everytime and is repeating a lot.

**Review by colm-quinn — COMMENTED** [review]
Great progress. I've left some improvement comments - mostly docs & naming, also some cleanup to make the future PR's easier by centralising changes in utl for reading the profile, if possible.

File: examples/x509-authentication-db-user/x509-authentication-db-user.json (hunk 2 above)
Comment (Adelmar92) [reply examples/x509-authentication-db-user/x509-authentication-db-user.json:8]: ahh yeah is a good idea

File: cfn-resources/serverless-instance/cmd/resource/resource.go (hunk 4 above)
Comment (Adelmar92) [reply cfn-resources/serverless-instance/cmd/resource/resource.go]: i was thinking about that, but all the currentmodels are different, we might need to do some overengineering to do this on the NewMongoClient

**Comment by Zuhairahmed** [comment]
@Adelmar92 can you also review GitHub automated errors and correct on this PR?

File: cfn-resources/serverless-instance/cmd/resource/resource.go (hunk 4 above)
Comment (colm-quinn) [reply cfn-resources/serverless-instance/cmd/resource/resource.go]: Ack, we can reconsider in the future when all the CFN changes are in place and review usage in a single block.

**Review by colm-quinn — DISMISSED** [review]
LGTM!

**Review by colm-quinn — APPROVED** [review]
LGTM!

**Pull request merged**